freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/HostKeytabService.java

```diff
@@ -0,0 +1,81 @@
+package com.sequenceiq.freeipa.kerberosmgmt.v1;
+
+import static com.sequenceiq.freeipa.kerberosmgmt.v1.KeytabCommonService.PRIVILEGE_DOES_NOT_EXIST;
+import static com.sequenceiq.freeipa.kerberosmgmt.v1.KeytabCommonService.ROLE_NOT_ALLOWED;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.service.secret.model.StringToSecretResponseConverter;
+import com.sequenceiq.freeipa.api.v1.kerberosmgmt.model.HostKeytabRequest;
+import com.sequenceiq.freeipa.api.v1.kerberosmgmt.model.HostKeytabResponse;
+import com.sequenceiq.freeipa.client.FreeIpaClient;
+import com.sequenceiq.freeipa.client.FreeIpaClientException;
+import com.sequenceiq.freeipa.client.model.Host;
+import com.sequenceiq.freeipa.entity.KeytabCache;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.kerberosmgmt.exception.KeytabCreationException;
+import com.sequenceiq.freeipa.service.freeipa.FreeIpaClientFactory;
+
+@Service
+public class HostKeytabService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HostKeytabService.class);
+
+    @Inject
+    private KeytabCommonService keytabCommonService;
+
+    @Inject
+    private StringToSecretResponseConverter secretResponseConverter;
+
+    @Inject
+    private FreeIpaClientFactory freeIpaClientFactory;
+
+    @Inject
+    private KerberosMgmtRoleComponent roleComponent;
+
+    public HostKeytabResponse generateHostKeytab(HostKeytabRequest request, String accountId) throws FreeIpaClientException {
+        LOGGER.debug("Request to generate host keytab: {}", request);
+        Stack freeIpaStack = keytabCommonService.getFreeIpaStack(request.getEnvironmentCrn(), accountId);
+        FreeIpaClient ipaClient = freeIpaClientFactory.getFreeIpaClientForStack(freeIpaStack);
+        if (!roleComponent.privilegesExist(request.getRoleRequest(), ipaClient)) {
+            throw new KeytabCreationException(PRIVILEGE_DOES_NOT_EXIST);
```

Comment:
- `KerberosMgmtRoleComponent#privilegesExist` is pretty well logged, so we would see from there everything, as exception mapper would log that an exception was thrown here, but if you still think it's useful, I can add a log here. Also this code is just moved, not new.
- right, some of the exceptions should be replaced with `BadRequestException`, but I haven't changed this part of the code. Almost all exception is transformed into `KeytabCreationException`. I think this could be a separate task to refactor it.

Reply:
- if it is well logged then there is no need to add an extra log.
- yes, it would make sense to create a refactor task.